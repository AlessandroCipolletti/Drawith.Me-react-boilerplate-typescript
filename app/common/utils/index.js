export function parseColorRgbStringToRgba (color) {
  color = color.substring(4).replace(")", "").replace(" ", "").split(",")
  return {
    r: parseInt(color[0]),
    g: parseInt(color[1]),
    b: parseInt(color[2]),
    a: color[3] ? parseInt(color[3]) : 1,
  }
}

export function parseColorHexToRgb (hex) {
  return {
    r: parseInt(hex.substring(0, 2), 16),
    g: parseInt(hex.substring(2, 4), 16),
    b: parseInt(hex.substring(4, 6), 16),
  }
}

export function parseColorRgbToHex (r, g, b) {
  const hex = ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1)
  return `#${hex}`
}

export function parseColorHexToRgbaString (colorHex, alpha = 1) {
  const r = parseInt(colorHex.substr(1, 2), 16)
  const g = parseInt(colorHex.substr(3, 2), 16)
  const b = parseInt(colorHex.substr(5, 2), 16)
  return `rgba(${r}, ${g}, ${b}, ${alpha})`
}
