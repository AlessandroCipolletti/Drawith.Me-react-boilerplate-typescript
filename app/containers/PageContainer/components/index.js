import InfoPopup from './InfoPopup'
import AppHeader from './AppHeader'

export {
  InfoPopup,
  AppHeader,
}
