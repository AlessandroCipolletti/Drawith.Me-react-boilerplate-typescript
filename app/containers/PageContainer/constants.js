import Theme from 'common/Theme'

const HEADER_HEIGHT = Theme.spacing.size7

export default HEADER_HEIGHT
