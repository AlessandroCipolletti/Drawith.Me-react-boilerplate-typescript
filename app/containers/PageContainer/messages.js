import { defineMessages } from 'react-intl'

export const scope = 'app.containers.PageContainer'

export default defineMessages({
  appName: {
    id: `${scope}.appName`,
    defaultMessage: 'Drawith.Me',
  },
  appDescription: {
    id: `${scope}.appDescription`,
    defaultMessage: 'Share your creativity on the world',
  },
})
