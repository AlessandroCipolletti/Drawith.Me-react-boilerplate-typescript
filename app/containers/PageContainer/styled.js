import styled from 'styled-components'
import Theme from 'common/Theme'

import headerBG from 'images/UI/PageContainer/headerBG.png'
import logo from 'images/UI/PageContainer/logo.png'

const headerHeight = Theme.spacing.size7

export const Wrapper = styled.div`
  position: fixed;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
`

export const AppHeader = styled.div`
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: ${headerHeight};
  border-bottom: solid 1px ${props => props.theme.palette.gray[6]};
  background-image: url(${logo}), url(${headerBG});
  background-size: auto 90%, contain;
  background-repeat: no-repeat, repeat-x;
  background-position: center, 0% 0%;
  z-index: 1000;
`

AppHeader.Button = styled.div`
  top: 0px;
  width: ${headerHeight};
  height: ${headerHeight};
  float: ${props => props.float || 'left'};
  box-sizing: border-box;
  padding: ${props => props.theme.spacing.size1};
  color: ${props => props.theme.palette.white};
  font-size: ${props => props.theme.fontSizing.size6};
  line-height: ${headerHeight};
  cursor: pointer;
`

export const PageContent = styled.div`
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  padding-top: ${headerHeight};
`
