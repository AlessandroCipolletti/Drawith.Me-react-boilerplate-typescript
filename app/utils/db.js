import Dexie from 'dexie'
// /* eslint-disable no-unused-vars */
// import setGlobalVars from 'indexeddbshim'
// setGlobalVars()

const db = new Dexie('DrawithMe.drawings')
db.version(1).stores({
  user: 'id, publicName, firstName, lastName',
  drawings: '++id, userId, folderId, state, createTime, updateTime, previewPath, imagePath, minX, minY, maxX, maxY, width, height',
  folders: '++id, name, order',
})

export default db
