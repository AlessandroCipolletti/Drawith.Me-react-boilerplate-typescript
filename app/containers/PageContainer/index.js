import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { Helmet } from 'react-helmet'
import { injectIntl, intlShape } from 'react-intl'

import messages from './messages'

export function PageContainer({
  name, children, intl,
}) {
  return (
    <div>
      <Helmet>
        <title>{name} - {intl.formatMessage(messages.appName)}</title>
        <meta name="description" content={intl.formatMessage(messages.appDescription)} />
      </Helmet>
      {children}
    </div>
  )
}

PageContainer.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  intl: intlShape.isRequired,
}

export default compose(
  injectIntl,
)(PageContainer)
