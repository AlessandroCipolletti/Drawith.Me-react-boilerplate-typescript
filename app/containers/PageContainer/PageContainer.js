import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { Helmet } from 'react-helmet'
import { injectIntl, intlShape } from 'react-intl'

import messages from './messages'

import { Wrapper, PageContent } from './styled'
// import { InfoPopup, AppHeader } from './components'
import InfoPopup from './components/InfoPopup'
import AppHeader from './components/AppHeader'

const PageContainer = ({
  name, headerButtons = [], children, intl,
}) => {
  const [infoPopupVisible, setInfoPopupVisible] = React.useState(false)
  const showInfoPopup = React.useCallback(() => {
    setInfoPopupVisible(true)
  }, [])
  const hideInfoPopup = React.useCallback(() => {
    setInfoPopupVisible(false)
  }, [])

  return (
    <div>
      <Helmet>
        <title>{name} - {intl.formatMessage(messages.appName)}</title>
        <meta name="description" content={intl.formatMessage(messages.appDescription)} />
      </Helmet>
      {infoPopupVisible && <InfoPopup callback={hideInfoPopup} />}
      <Wrapper>
        <AppHeader
          headerButtons={headerButtons}
          showInfoPopup={showInfoPopup}
        />
        <PageContent>
          {children}
        </PageContent>
      </Wrapper>
    </div>
  )
}

PageContainer.propTypes = {
  name: PropTypes.string.isRequired,
  headerButtons: PropTypes.array,
  children: PropTypes.any.isRequired,
  intl: intlShape.isRequired,
}

export default compose(
  injectIntl,
)(PageContainer)
