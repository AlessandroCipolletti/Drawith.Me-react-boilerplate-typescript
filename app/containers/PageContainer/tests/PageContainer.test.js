import React from 'react'
import { render } from 'react-testing-library'

// Import de Intl Provider
import { IntlProvider, injectIntl } from 'react-intl'
import { compose } from 'redux'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/Theme'
import { DEFAULT_LOCALE } from 'utils/i18n'

import PageContainer from '../PageContainer'

describe('<PageContainer />', () => {

  const ComponentWithIntl = compose(injectIntl)(PageContainer)
  const renderComponent = (props = {}) => (
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ThemeProvider theme={Theme}>
          <ComponentWithIntl {...props}>
            Content
          </ComponentWithIntl>
        </ThemeProvider>
      </IntlProvider>,
    )
  )

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    const name = 'name'

    renderComponent({ name })

    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect log an error in console if called without name prop', () => {
    const spy = jest.spyOn(global.console, 'error')

    renderComponent()

    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('Should render and match the snapshot with default props', () => {
    const name = 'name'

    const {
      container: { firstChild },
    } = renderComponent({ name })

    expect(firstChild).toMatchSnapshot()
  })

})
