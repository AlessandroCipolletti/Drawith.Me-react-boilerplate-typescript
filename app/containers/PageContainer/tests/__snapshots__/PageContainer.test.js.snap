// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> Should render and match the snapshot with default props 1`] = `
<div>
  <div
    class="styled__Wrapper-lvdcr6-0 jJmZCw"
  >
    <div
      class="styled__Wrapper-sc-1a6eih8-0 jXWNUQ"
    >
      <div
        class="styled__HeaderButton-sc-1a6eih8-1 fTDkle"
      >
        <div
          class="Icon__Wrapper-xawg7a-0 hKOtws"
          color="#22252a"
          role="button"
          size="32"
          tabindex="0"
        />
      </div>
    </div>
    <div
      class="styled__PageContent-lvdcr6-1 fptcef"
    >
      Content
    </div>
  </div>
</div>
`;
